lib/process-services-cloud/src/lib/group-cloud/components/group-cloud.component.ts

```diff
@@ -0,0 +1,254 @@
+import { trigger, state, style, transition, animate } from '@angular/animations';
+import { Observable, of, BehaviorSubject } from 'rxjs';
+import { GroupModel, GroupSearchParam } from '../models/group.model';
+import { GroupCloudService } from '../services/group-cloud.service';
+import { debounceTime } from 'rxjs/internal/operators/debounceTime';
+import { distinctUntilChanged, switchMap, flatMap, mergeMap, filter, tap } from 'rxjs/operators';
+
+@Component({
+    selector: 'adf-cloud-group',
+    templateUrl: './group-cloud.component.html',
+    styleUrls: ['./group-cloud.component.scss'],
+    animations: [
+        trigger('transitionMessages', [
+            state('enter', style({ opacity: 1, transform: 'translateY(0%)' })),
+            transition('void => enter', [
+                style({ opacity: 0, transform: 'translateY(-100%)' }),
+                animate('300ms cubic-bezier(0.55, 0, 0.55, 0.2)')
+            ])
+        ])
+    ],
+    encapsulation: ViewEncapsulation.None
+})
+export class GroupCloudComponent implements OnInit {
+
+    static MODE_SINGLE = 'single';
+    static MODE_MULTIPLE = 'multiple';
+    static NO_MATCH_FOUND = 'match';
+    static NO_ROLE_MAPPING = 'rolemapping';
+
+    @ViewChild('groupInput') groupInput: ElementRef<HTMLInputElement>;
+
+    /** (required) Name of the app. */
+    @Input()
+    applicationName: string;
+
+    /** Mode of the group selection (single/multiple) */
+    @Input()
+    mode: string = GroupCloudComponent.MODE_SINGLE;
+
+    @Input()
+    preSelectGroups: GroupModel[];
+
+    /** Emitted when a group is selected. */
+    @Output()
+    selectGroup: EventEmitter<GroupModel> = new EventEmitter<GroupModel>();
+
+    /** Emitted when a group is removed. */
+    @Output()
+    removeGroup: EventEmitter<GroupModel> = new EventEmitter<GroupModel>();
+
+    /** Emitted when an error occurs. */
+    @Output()
+    error: EventEmitter<boolean> = new EventEmitter<boolean>();
+
+    private _selectedGroups: GroupModel[] = [];
+
+    private selectedGroups: BehaviorSubject<GroupModel[]>;
+
+    private searchGroups: BehaviorSubject<GroupModel[]>;
+
+    applicationId: string;
+
+    groups$: Observable<GroupModel[]>;
+
+    selectedGroups$: Observable<GroupModel[]>;
+
+    searchGroup: FormControl = new FormControl();
+
+    _subscriptAnimationState = 'enter';
+
+    filtered = [];
+
+    showHint = false;
+    errorMessage: string;
+    private validationMessages = { match: 'ADF_CLOUD_GROUPS.ERROR.NOT_FOUND', rolemapping: 'ADF_CLOUD_GROUPS.ERROR.ROLE_MAPPING' };
+
+    constructor(private groupService: GroupCloudService) {
+        this.selectedGroups = new BehaviorSubject<GroupModel[]>(this._selectedGroups);
+        this.searchGroups = new BehaviorSubject<GroupModel[]>(this._selectedGroups);
+        this.selectedGroups$ = this.selectedGroups.asObservable();
+        this.groups$ = this.searchGroups.asObservable();
+    }
+
+    ngOnInit() {
+        this.getClientId();
```

Comment:
The two methods (getClientId and initSearch) are async so it could happen that the second one will complete before the first one.

Create an async method 

async initSearch() {
this.clientId = await getClientId(); 
search();
}